hail/src/main/scala/is/hail/linalg/LAPACK.scala

```diff
@@ -0,0 +1,87 @@
+package is.hail.linalg
+
+import java.lang.reflect.Method
+
+import com.sun.jna.{FunctionMapper, Library, Native, NativeLibrary, Pointer}
+import com.sun.jna.ptr.{DoubleByReference, IntByReference}
+
+import scala.util.{Failure, Success, Try}
+import is.hail.utils._
+
+class UnderscoreFunctionMapper extends FunctionMapper {
+  override def getFunctionName(library: NativeLibrary, method: Method): String = {
+    method.getName() + "_"
+  }
+}
+
+object TryAll {
+  def apply[K](f: => K): Try[K] =
+    try {
+      Success(f)
+    } catch {
+      case e: Throwable => Failure(e)
+    }
+}
+
+object LAPACK {
+  lazy val libraryInstance = {
+    val standard = Native.loadLibrary("lapack", classOf[LAPACKLibrary]).asInstanceOf[LAPACKLibrary]
+
+    versionTest(standard) match {
+      case Success(version) =>
+        log.info(s"Imported LAPACK version ${version} with standard names")
+        standard
+      case Failure(exception) =>
+        val underscoreAfterMap = new java.util.HashMap[String, FunctionMapper]()
+        underscoreAfterMap.put(Library.OPTION_FUNCTION_MAPPER, new UnderscoreFunctionMapper)
+        val underscoreAfter = Native.loadLibrary("lapack", classOf[LAPACKLibrary], underscoreAfterMap).asInstanceOf[LAPACKLibrary]
+        versionTest(underscoreAfter) match {
+          case Success(version) =>
+            log.info(s"Imported LAPACK version ${version} with underscore names")
+            underscoreAfter
+          case Failure(exception) =>
+            throw exception
+        }
+    }
+  }
+
+  // Return info
+  def dgeqrf(M: Int, N: Int, A: Long, LDA: Int, TAU: Long, WORK: Long, LWORK: Int): Int = {
+    val mInt = new IntByReference(M)
+    val nInt = new IntByReference(N)
+    val LDAInt = new IntByReference(LDA)
+    val LWORKInt = new IntByReference(LWORK)
+    val infoInt = new IntByReference(1)
+    libraryInstance.dgeqrf(mInt, nInt, A, LDAInt, TAU, WORK, LWORKInt, infoInt)
+    infoInt.getValue()
+  }
+
+  // Return info
```

Comment:
I think this is pretty clear from the `infoInt.getValue` so I wouldn't include it.